ignite/handlers/state_param_scheduler.py

```diff
@@ -0,0 +1,390 @@
+import numbers
+from bisect import bisect_right
+from typing import Any, List, Tuple, Union
+
+from ignite.engine import CallableEventWithFilter, Engine, Events, EventsList
+from ignite.handlers import BaseParamScheduler
+
+
+class StateParamScheduler(BaseParamScheduler):
+    """An abstract class for updating an engine state parameter values during training.
+
+    Args:
+        param_name: name of parameter to update.
+        save_history: whether to log the parameter values to
+            `engine.state.param_history`, (default=False).
+
+    Note:
+        Parameter scheduler works independently of the internal state of the attached engine.
+        More precisely, whatever the state of the engine (newly created or used by another scheduler) the scheduler
+        sets defined absolute values.
+
+    .. versionadded:: 0.6.0
+
+    """
+
+    def __init__(
+        self, param_name: str, save_history: bool = False,
+    ):
+        super(StateParamScheduler, self).__init__(param_name, save_history)
+
+    def attach(
+        self,
+        engine: Engine,
+        event: Union[str, Events, CallableEventWithFilter, EventsList] = Events.ITERATION_COMPLETED,
+    ) -> None:
+        """Attach the handler to the engine. Once the handler is attached, the ``Engine.state`` will have a new
+        attribute with the name ``param_name``. Then the current value of the parameter can be retrieved from
+        ``Engine.state`` when the engine is running.
+
+        Args:
+            engine: trainer to which the handler will be attached.
+            event: trigger ``param_name`` value update.
+
+        """
+        if hasattr(engine.state, self.param_name):
+            raise ValueError(
```

Comment:
@fco-dv could you add a test ?

Reply:
I've added a test for that `test_state_param_asserts`